x/subspaces/legacy/v7/store.go

```diff
@@ -0,0 +1,38 @@
+package v7
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/v5/x/subspaces/types"
+)
+
+// MigrateStore migrates the store from version 6 to version 7.
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec) error {
+	// Add user allowances to expiring queue
+	addAllowancesToExpiringQueue(ctx, storeKey, cdc, types.UserAllowancePrefix)
+
+	// Add group allowances to expiring queue
+	addAllowancesToExpiringQueue(ctx, storeKey, cdc, types.GroupAllowancePrefix)
+	return nil
+}
+
+func addAllowancesToExpiringQueue(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec, allowancePrefix []byte) {
+	store := ctx.KVStore(storeKey)
+	allowanceStore := prefix.NewStore(store, allowancePrefix)
+	iterator := allowanceStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var grant types.Grant
+		cdc.MustUnmarshal(iterator.Value(), &grant)
+
+		expiration := grant.GetExpiration()
+		if expiration != nil {
+			allowanceKey := allowancePrefix
+			allowanceKey = append(allowanceKey, iterator.Key()...)
```

Comment:
These lines can be combined
```suggestion
			allowanceKey := append(allowancePrefix, iterator.Key()...)
```

Reply:
It would cause the linter error: `appendAssign: append result not assigned to the same slice (gocritic)`.
https://github.com/desmos-labs/desmos/actions/runs/5552100585/jobs/10139013465#step:5:146

Reply:
What about something like this?
```go
allowanceKey := append([]byte(allowanceKey...), iterator.Key()...)
```

Reply:
It would be another linter problem: 
``
x/subspaces/legacy/v7/store.go:33:33: unnecessary conversion (unconvert)
                        allowanceKey := append([]byte(allowancePrefix), iterator.Key()...)
``

Reply:
Ok then we can leave it as it is

Reply:
I refer to the similar codes which uses `append([]byte(nil), prefix)` pattern. I think we can apply it. 

https://github.com/desmos-labs/desmos/blob/master/x/profiles/keeper/grpc_query.go#L179